os/drivers/error_report/error_report_drv.c

```diff
@@ -0,0 +1,301 @@
+		if ((tcbptr->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD) {
+			return (unsigned long)e.pthread;
+		} else {
+			return (unsigned long)e.main;
+		}
+	}
+	return 0;
+}
+
+static void prv_tcb_handler(FAR struct tcb_s *tcb, void *args)
+{
+	int pos = 0;
+	if (tcb->pid < 5) {
+		return;
+	}
+	if ((tcb->task_state >= TSTATE_WAIT_SEM) && (tcb->task_state < NUM_TASK_STATES)) {
+		int ret;
+		struct cpuload_s cpu;
+		ret = clock_cpuload(tcb->pid, 0, &cpu);
+		if (ret != OK) {
+			return;
+		}       
+		/* Hash pid over the waitq range */
+		pos = tcb->pid % CONFIG_MAX_TASKS;
+		if ((g_thd_waitq[pos].pid == tcb->pid)) {
+			if (g_thd_waitq[pos].task_state == tcb->task_state) {
+				if (g_thd_waitq[pos].active_cpu == cpu.active) {
+					int threshold;
+					g_thd_waitq[pos].count++;
+					if (prv_is_reserved(tcb->name) == 1) {
+						threshold = INFINITE_CHECK_HIGH_THRESHOLD;
+					}
+					else {
+						threshold = CONFIG_ERROR_REPORT_INFINITE_CHECK_THRESHOLD;
+					}
+					
+					nwerr_vdbg("Inc wait count for pid %d\n", tcb->pid);
+					if (g_thd_waitq[pos].count >= threshold) {
+						// Add code for error report addition
+						uint32_t task_addr;
+						uint32_t saved_pc;
+						entry_t e = tcb->entry;
+						if ((tcb->flags & TCB_FLAG_TTYPE_MASK) == TCB_FLAG_TTYPE_PTHREAD) {
+							task_addr = (unsigned long)e.pthread;
+						} else {
+							task_addr = (unsigned long)e.main;
+						}					
+						saved_pc = tcb->xcp.regs[13];
+						nwerr_vdbg("pid %d blocked\n", tcb->pid);
+						g_create_err_rec(tcb->pid, tcb->task_state, task_addr);
+						/* Reset the count, indicating that is free for insert */
+						DELETE_INFINITE_WAITQ(pos);
+					}
+				} else {
+					/* The thread was running in between, and blocked again */
+					INSERT_INFINITE_WAITQ(pos, tcb, cpu);
+				}				
+			} else {
+				/* Ideally does not happen, but we account for it anyway */
+				INSERT_INFINITE_WAITQ(pos, tcb, cpu);
+			}
+		} 
+		else if (!g_thd_waitq[pos].pid) {
+			/* Add new entry */
+			INSERT_INFINITE_WAITQ(pos, tcb, cpu);
+		}
+		else {
+			/* Collision resolution, find the next empty slot */
+			int i;
+			for (i = pos+1; i != pos; i = (i+1)%CONFIG_MAX_TASKS) {
+				if (g_thd_waitq[i].pid == 0) {
+					/* Found an empty slot, insert values here */
+					INSERT_INFINITE_WAITQ(i, tcb, cpu);
+					break;
+				}
+			}
+			if (i == pos) {
+				nwerrdbg("Cannot find a slot to insert values\n");
+			}
+		}		
+	}
+	else {
+		/* Maybe the pid was blocked during the 
+		 * last sampling instance, and 
+		 * is active again. Reset the data 
+		 * in this case. 
+		 */
+		/* Hash pid over the waitq range */
+		pos = tcb->pid % CONFIG_MAX_TASKS;
+		if ((g_thd_waitq[pos].pid == tcb->pid)) {
+			/* Reset the count, indicating that is free for insert */
+			DELETE_INFINITE_WAITQ(pos);
+		} else if (g_thd_waitq[pos].pid) {
+			/* Collision resolution, find the slot using linear search */
+			int i;
+			for (i = pos+1; i != pos; i = (i+1)%CONFIG_MAX_TASKS) {
```

Comment:
needs spaces around `+` and `%`

Reply:
Addressed in a similar, yet different comment.